app/Models/Mship/Account.php

```diff
@@ -437,31 +457,25 @@ public function getFullyDefinedAttribute()
      */
     public function isValidDisplayName($displayName)
     {
-        $allowedNames = collect();
-        $allowedNames->push($this->name);
-        $allowedNames->push($this->real_name);
-
-        if ($this->networkDataAtcCurrent) {
-            $allowedNames->push($this->name.' - '.$this->networkDataAtcCurrent->callsign);
-            $allowedNames->push($this->real_name.' - '.$this->networkDataAtcCurrent->callsign);
-        }
-
-        return $allowedNames->filter(function ($item, $key) use ($displayName) {
+        return $this->allowedNames(true)->filter(function ($item, $key) use ($displayName) {
             return strcasecmp($item, $displayName) == 0;
         })->count() > 0;
     }
 
     public function isPartiallyValidDisplayName($displayName)
     {
-        $allowedNames = collect();
-        $allowedNames->push($this->name);
-        $allowedNames->push($this->real_name);
-
-        return $allowedNames->filter(function ($item, $key) use ($displayName) {
+        return $this->allowedNames()->filter(function ($item, $key) use ($displayName) {
             return strstr(strtolower($displayName), strtolower($item)) != false;
```

Comment:
So basically these would be true? (!= false) 😉

Reply:
I'm note sure I understand what you mean?

Reply:
Evaluating != false is the same as something being true, but then I forgot strstr() returns false if nothing is found. Gotta love inconsistent return types.